File: src/test/java/io/vertx/support/CollectionFactory.java
package io.vertx.support;

import io.vertx.codegen.annotations.VertxGen;

import java.util.HashMap;
import java.util.Map;

/**
 * @author <a href="mailto:julien@julienviet.com">Julien Viet</a>
 */
@VertxGen
public interface CollectionFactory {

  static Map<String, String> createMap() {
    Map<String, String> map = new HashMap<>();
    map.put("foo", "foo_value");
    return map;
  }

  static Map<String, String> wrapMap(Map<String, String> map) {
    return map;
  }
}

File: src/test/resources/io/vertx/scala/support/CollectionFactory.scala
package io.vertx.scala.support

/**
 * @author <a href="mailto:jochen.mader@codecentric.de">Jochen Mader</a>
 */
object CollectionFactory {
  def createMap() = {
    Map("foo" -> "foo_value")
  }
  def wrapMap(map: Map[String,String]) = {
    map
  }
}
